src/tasks/FileUploadUtil/FileObjectClasses/StreamUpload.ts

```diff
@@ -0,0 +1,82 @@
+import { Readable } from "stream";
+
+import { GraphClientError } from "../../../GraphClientError";
+import { Range } from "../../../Range";
+import { FileObject } from "../../LargeFileUploadTask";
+export class StreamUpload implements FileObject {
+	content: Readable;
+	name: string;
+	size: number;
+	public constructor(content: Readable, name: string, size: number) {
+		if (!content || !name || !size) {
+			throw new GraphClientError("Please provide the Readable Stream content, name of the file and size of the file");
+		}
+		this.content = content;
+		this.size = size;
+		this.name = name;
+	}
+
+	/**
+	 * @public
+	 * Slices the file content to the given range
+	 * @param {Range} range - The range value
+	 * @returns The sliced file part
+	 */
+	public async sliceFile(range: Range): Promise<ArrayBuffer | Blob | Buffer> {
+		const rangeSize = range.maxValue - range.minValue + 1;
+		/* readable.readable Is true if it is safe to call readable.read(),
+		 * which means the stream has not been destroyed or emitted 'error' or 'end'
+		 */
+		if (this.content.readable) {
```

Comment:
as the content property is public (default in typescript), someone could set it as null/undefined and this would derail.
You need to either change the access modifier, or add a readonly, or defend here.